gst/nnstreamer/tensor_source/tensor_src_iio.c

```diff
@@ -217,28 +282,390 @@ gst_tensor_src_iio_class_init (GstTensorSrcIIOClass * klass)
+  g_free (prop->name);
+  g_free (prop->generic_name);
+  g_free (prop->base_dir);
+  g_free (prop->base_file);
+}
+
+/**
+ * @brief initialize GstTensorSrcIIODeviceProperties structure
+ * TODO: verify that this is needed
+ */
+static void
+gst_tensor_src_iio_device_properties_init (GstTensorSrcIIODeviceProperties *
+    prop)
+{
+  prop->name = NULL;
+  prop->base_dir = NULL;
+  prop->id = -1;
+}
+
 /**
  * @brief initialize tensor_src_iio element.
  */
 static void
 gst_tensor_src_iio_init (GstTensorSrcIIO * self)
 {
+  // TODO: verify where locking is needed
   g_mutex_init (&self->mutex);
 
   /** init properties */
-  /** already set using default values */
+  self->configured = FALSE;
+  self->channels = DEFAULT_PROP_STRING;
+  self->channels_enabled = CHANNELS_ENABLED_AUTO;
+  gst_tensor_src_iio_device_properties_init (&self->trigger);
+  gst_tensor_src_iio_device_properties_init (&self->device);
+  self->silent = DEFAULT_PROP_SILENT;
+  self->buffer_capacity = DEFAULT_BUFFER_CAPACITY;
+  self->sampling_frequency = DEFAULT_FREQUENCY;
 
   return;
 }
 
 /**
- * @brief set properties of tensor_src_iio
+ * @brief check if device/trigger with the given name exists
+ * @return >=0 if OK, represents device/trigger number
+ *         -1  if returned with error
+ */
+static guint
+gst_tensor_src_iio_get_id_by_name (const gchar * dir_name, const gchar * name,
+    const gchar * prefix)
+{
+  DIR *dptr = NULL;
+  GError *error = NULL;
+  struct dirent *dir_entry;
+  gchar *filename = NULL;
+  guint64 id = -1;
+  gchar *file_contents = NULL;
+  guint ret = -1;
+
+  dptr = opendir (dir_name);
+  if (G_UNLIKELY (NULL == dptr)) {
+    GST_ERROR ("No IIO %s available.", prefix);
+    goto error_close_dir;
+  }
+
+  while ((dir_entry = readdir (dptr)) != NULL) {
+    // check for prefix and the next digit should be a number
+    if (g_str_has_prefix (dir_entry->d_name, prefix) &&
+        g_ascii_isdigit (dir_entry->d_name[strlen (prefix)])) {
+
+      id = g_ascii_strtoll (dir_entry->d_name + strlen (prefix), NULL, 10);
+      filename =
+          g_strdup_printf ("%s%s/%s", dir_name, dir_entry->d_name, NAME_FILE);
+
+      if (!g_file_get_contents (filename, &file_contents, NULL, &error)) {
+        GST_ERROR ("Unable to read %s, error: %s\n.", filename, error->message);
+        g_error_free (error);
+        goto error_free_filename;
+      }
+      g_free (filename);
+
+      if (!g_strcmp0 (file_contents, name)) {
+        ret = id;
+        g_free (file_contents);
+        break;
+      }
+      g_free (file_contents);
+    }
+  }
+
+  closedir (dptr);
+  return ret;
+
+error_free_filename:
+  g_free (filename);
+
+error_close_dir:
+  closedir (dptr);
+  return ret;
+}
+
+/**
+ * @brief get type info about the channel from the string
+ */
+static gboolean
+gst_tensor_src_iio_set_channel_type (GstTensorSrcIIOChannelProperties * prop,
+    const gchar * file_contents)
+{
+  gchar endianchar, signchar;
```

Comment:
initialize these two before using them although sscanf is supposed to set them.

Reply:
Done.